BCH26.c

```diff
@@ -1,6 +1,26 @@
-#include <stdlib.h>
+/*
+ *      bch26.c -- BCH(26,16) Decoder
+ *      author: Alex L Manstein (alex.l.manstein@gmail.com)
+ *      Copyright (C) 2020
+ *
+ *	This program is free software; you can redistribute it and/or modify
+ *	it under the terms of the GNU General Public License as published by
+ *	the Free Software Foundation; either version 2 of the License, or
+ *	(at your option) any later version.
+ *
+ *	This program is distributed in the hope that it will be useful,
+ *	but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *	GNU General Public License for more details.
+ *
+ *	You should have received a copy of the GNU General Public License
+ *	along with this program; if not, write to the Free Software
+ *	Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
```

Comment:
I'm not a lawyer, but I'm sure that attaching the GPL copyright notices to a piece of proprietary code does not make it automagically GPL-compatible. Permission is required from the original author to a) make an adaptation, and b) does not given attribution to her or him in the copyright notice of the adoption work.

Maybe you can ask the author to share the code under a proper license. Or, you might just adapt a GPL-compatible alternative, say, https://github.com/torvalds/linux/blob/master/lib/bch.c.